packages/core/helper-plugin/lib/src/components/RelationInput/RelationInput.js

```diff
@@ -114,28 +122,31 @@ export const RelationInput = ({
   );
 };
 
-const RelationTypeDef = PropTypes.shape({
-  id: PropTypes.number.isRequired,
-  isDraft: PropTypes.bool,
-  href: PropTypes.string,
-  title: PropTypes.string.isRequired,
-});
-
 const ReactQueryRelationResult = PropTypes.shape({
   data: PropTypes.shape({
-    pages: PropTypes.arrayOf(RelationTypeDef),
+    pages: PropTypes.arrayOf(
+      PropTypes.shape({
+        id: PropTypes.number.isRequired,
+        isDraft: PropTypes.bool,
+        href: PropTypes.string,
+        title: PropTypes.string.isRequired,
+      })
+    ),
   }),
   isLoading: PropTypes.bool.isRequired,
   isSuccess: PropTypes.bool.isRequired,
 });
 
-const ReactQuerySearchResult = PropTypes.shape({
-  data: PropTypes.shape({
-    pages: PropTypes.arrayOf(RelationTypeDef),
-  }),
-  isLoading: PropTypes.bool.isRequired,
-  isSuccess: PropTypes.bool.isRequired,
-});
+const ReactQuerySearchResult = PropTypes.arrayOf(
+  PropTypes.shape({
+    label: PropTypes.string,
+    value: {
+      id: PropTypes.number,
+      name: PropTypes.string,
+      publishedAt: PropTypes.string,
```

Comment:
@gu-stav The current data we receive for search results has this format, I checked with Pierre and it seems it'll be the same.
The label key comes from a map we do in SelectWrapper though, not sure why and if we should keep it.
I guess we'll receive `isLoading` and `isSuccess` too?

Reply:
The search result will look like I've defined them:

```js
const ReactQuerySearchResult = PropTypes.shape({
  data: PropTypes.shape({
    pages: PropTypes.arrayOf(RelationTypeDef),
  }),
  isLoading: PropTypes.bool.isRequired,
  isSuccess: PropTypes.bool.isRequired,
});
```

You can read about it here: https://tanstack.com/query/v4/docs/guides/infinite-queries

The props of `RelationTypeDef` are not yet fixed and can change.

Reply:
good to know thank you!
I'm still wondering if data will have the same format for `ReactQuerySearchResult` and `ReactQueryRelationResult` because Option component need to specifically know about the data structure to display the right title - do we leave that for later?

<img width="679" alt="image" src="https://user-images.githubusercontent.com/71838159/186112211-29bc02b0-2cdf-4182-9038-32d5efc03da4.png">

Reply:
I'm not sure about it either. I think for now you could assume both (search, existing) are different and I'm fine with splitting the prop-type for `RelationTypeDef`.

The overall format for `ReactQuerySearchResult` and `ReactQueryRelationResult` will look the same. I don't remember exactly how, but every `Option` has access to it's full data somewhere. I believe you could have a look at https://github.com/strapi/strapi/blob/main/packages/core/upload/admin/src/components/SelectTree/Option.js

If you want to finish this PR rather earlier than later you can leave that for now until we have the endpoint.

Reply:
indeed from what I've seen Option receives everything throught `props.data` that's how we access 
`const isDraft = isEmpty(get(props, 'data.value.publishedAt'))`

@petersg83 could you confirm or refute the data when searching for new relations would be: 
```
{
  results: [{
    id: 2,
    name,: 'Skate', // c'est le mainField
    publishedAt: null,
  }],
  pagination: {
    page: 1,
    pageSize: 10,
    total: 1,
  },
}
```

then @gu-stav I guess we could have:

```
const ReactQuerySearchResult = PropTypes.shape({
  data: PropTypes.shape({
    pages: PropTypes.arrayOf(
         PropTypes.shape({
            label: PropTypes.string,
            value: {
              id: PropTypes.number,
              name: PropTypes.string,
              publishedAt: PropTypes.string,
            },
         })
    ),
  }),
  isLoading: PropTypes.bool.isRequired,
  isSuccess: PropTypes.bool.isRequired,
});
```
again `label` comes from a map we do in `SelectWrapper` by using the name key, not sure why we need it and if we could/should remove it?

Reply:
I didn't read all the conversation but I confirm the data will have this format :)

Reply:
@petersg83 For search I don't need the `pagination` attribute at all and you can omit it, if that is easy, to make the responses lighter.

I'm not sure what do to about the main-field: whether it could be computed in the BE and send back (I'd prefer that) or whether we pull it out in the FE.

Ideally the type would look like 

```js
const ReactQuerySearchResult = PropTypes.shape({
  data: PropTypes.shape({
    pages: PropTypes.arrayOf(
         PropTypes.shape({
            id: PropTypes.number,
            mainField: PropTypes.string,
            publishedAt: PropTypes.string,
         })
    ),
  }),
  isLoading: PropTypes.bool.isRequired,
  isSuccess: PropTypes.bool.isRequired,
});
```

for me, @ronronscelestes. If that is not possible, I'd go for 

```js
const ReactQuerySearchResult = PropTypes.shape({
  data: PropTypes.shape({
    pages: PropTypes.arrayOf(
         PropTypes.shape({
            id: PropTypes.number,
            name: PropTypes.string,
            publishedAt: PropTypes.string,
         })
    ),
  }),
  isLoading: PropTypes.bool.isRequired,
  isSuccess: PropTypes.bool.isRequired,
});
```

and provide you with the name of the main field (`name` is this case).

Reply:
Yes, you will receive the latter version + the main field of the target content type somehow.

Reply:
**Note**: we'll iterate on it once we know more about the data structure, a checkbox was added to `feature/relations-main-view` PR